earthfile2llb/converter.go

```diff
@@ -2306,6 +2307,10 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	// Shell and debugger wrap.
 	prependDebugger := !opts.Locally
 	finalArgs = opts.shellWrap(finalArgs, extraEnvVars, opts.WithShell, prependDebugger, isInteractive)
+	if opts.NoCache {
+		// llb.IgnoreCache is not always enough; we will force a different cache key as a work-around
+		finalArgs = append(finalArgs, "#"+uuid.NewString())
```

Comment:
Interestingly simple... I hope this works well in all the weird cases. I can't think of a specific situation where it wouldn't.

Reply:
I had a similar solution in a wip for a while.
The difference being attaching a uuid (the same uuid actually) for any subsequent RUNs that come after a RUN --no-cache, just because it was unclear if the bug affected those subsequent commands as well.

Reply:
yeah I figured we needed a new one each time to support the weird case of multiple IF statements which had the same llb deps (see my comment under the main PR if you're curious).

Reply:
Yeah, the new uuid makes sense(I only saw your example after posting my comment).
My only question is whether the bug in this example
```
RUN --no-cache echo hello
RUN echo goodbye
````
would sometimes cache and not echo "goodbye" And if so, should we make sure to attach a (new) uuid for every RUN that comes after `RUN --no-cache`

Reply:
both will be executed @idodod :

```
/h/a/t/simple-no-cache+base | --> RUN --no-cache echo hello
/h/a/t/simple-no-cache+base | hello
/h/a/t/simple-no-cache+base | DOCKERHUB_MIRROR_AUTH_FROM_CLOUD_SECRETS=true
/h/a/t/simple-no-cache+base | --> RUN echo goodbye
/h/a/t/simple-no-cache+base | goodbye
```

however this is consistent with what v0.8.12 does.

Reply:
Right, I was under the assumption that that is what is supposed to happen and that usually happens, but wasn't sure if the bug (which is somewhat inconsistent) affects the scenario I described.

Reply:
yeah it sort of feels like it goes against buildkit's slow-cache idea, but I think it's fine.

If you break it up into two separate targets, e.g.

```
VERSION 0.8
FROM alpine
a:
  RUN --no-cache echo hello > msg
  SAVE ARTIFACT msg

b:
  COPY +a/msg .
  RUN echo goodbye
```

it doesn't run the `echo goodbye`:

```
              logbus | Setting organization "" and project ""
                  +b | --> FROM +base
              alpine | --> Load metadata alpine linux/amd64
                  +a | --> FROM +base
               +base | --> FROM alpine
               +base | [----------] 100% FROM alpine
                  +a | --> RUN --no-cache echo hello > msg
                  +b | *cached* --> COPY +a/msg ./
                  +b | *cached* --> RUN echo goodbye
              output | --> exporting outputs
```

which makes it look like it does hit the slowcache (which is also consistent with v0.8.12).